samples/compose/src/main/java/com/squareup/sample/compose/hellocomposebinding/HelloWorkflow.kt

```diff
@@ -0,0 +1,50 @@
+package com.squareup.sample.compose.hellocomposebinding
+
+import com.squareup.sample.compose.hellocomposebinding.HelloWorkflow.Rendering
+import com.squareup.sample.compose.hellocomposebinding.HelloWorkflow.State
+import com.squareup.sample.compose.hellocomposebinding.HelloWorkflow.State.Goodbye
+import com.squareup.sample.compose.hellocomposebinding.HelloWorkflow.State.Hello
+import com.squareup.workflow1.Snapshot
+import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.action
+import com.squareup.workflow1.parse
+
+object HelloWorkflow : StatefulWorkflow<Unit, State, Nothing, Rendering>() {
+  enum class State {
+    Hello,
+    Goodbye;
+
+    fun theOtherState(): State = when (this) {
+      Hello -> Goodbye
+      Goodbye -> Hello
+    }
+  }
+
+  data class Rendering(
+    val message: String,
+    val onClick: () -> Unit
+  )
+
+  private val helloAction = action {
+    state = state.theOtherState()
+  }
+
+  override fun initialState(
+    props: Unit,
+    snapshot: Snapshot?
+  ): State = snapshot?.bytes?.parse { source -> if (source.readInt() == 1) Hello else Goodbye }
+    ?: Hello
+
+  override fun render(
+    renderProps: Unit,
+    renderState: State,
+    context: RenderContext
+  ): Rendering {
+    return Rendering(
+      message = renderState.name,
+      onClick = { context.actionSink.send(helloAction) }
```

Comment:
I guess `context.eventHandler` isn't used here because you want `helloAction` to be a class property and not reinstantiated with each render pass?

Reply:
I don't even remember what `eventHandler` does lol. I know it's some kind of sugar, I guess I should use it in the sample…

Reply:
Ah, wow yea that's a lot of magic - i'd rather keep this simple.